packages/lesswrong/components/forumEvents/ForumEventPostPagePollSection.tsx

```diff
@@ -0,0 +1,139 @@
+    }
+  },
+  heading: {
+    fontSize: 24,
+    fontWeight: 700,
+    lineHeight: 'normal',
+    marginTop: 0,
+    marginBottom: 8,
+  },
+  description: {
+    fontSize: 14,
+    fontWeight: 500,
+    lineHeight: '20px',
+    marginBottom: 20,
+    '& a': {
+      textDecoration: "underline",
+      textUnderlineOffset: '2px',
+    }
+  },
+  pollArea: {
+    paddingTop: 20,
+    borderRadius: theme.borderRadius.default,
+    '& .ForumEventPoll-question': {
+      fontSize: 24,
+    },
+    '& .ForumEventPoll-questionFootnote': {
+      fontSize: 16,
+    }
+  },
+});
+
+const announcementPostUrl = 'https://forum.effectivealtruism.org/posts/PeBNdpoRSq59kAfDW/announcing-ai-welfare-debate-week-july-1-7'
+
+/**
+ * This is used on the post page to display the current forum event's poll,
+ * and allow users to update their vote to show that the post changed their minds.
+ * This uses the theme name to set the root element's colors so you should NoSSR it.
+ */
+export const ForumEventPostPagePollSection = ({postId, classes}: {
+  postId: string,
+  classes: ClassesType<typeof styles>,
+}) => {
+  const {params} = useLocation();
+  const {currentForumEvent} = useCurrentForumEvent();
+  const currentUser = useCurrentUser()
+  const hasVoted = getForumEventVoteForUser(currentForumEvent, currentUser) !== null
+  const themeOptions = useConcreteThemeOptions()
+
+  const {document: post} = useSingle({
+    collectionName: "Posts",
+    fragmentName: "PostsDetails",
+    documentId: params._id,
+    skip: !hasForumEvents || !params._id || !currentForumEvent?.tagId || !currentForumEvent?.includesPoll,
+  });
+
+  // Only show this section for forum events that have a poll
+  if (!currentForumEvent || !post || !currentForumEvent.includesPoll) {
+    return null;
+  }
+
+  // Only show this section for posts tagged with the event tag
+  const relevance = post?.tagRelevance?.[currentForumEvent.tagId] ?? 0;
+  if (relevance < 1) {
+    return null;
+  }
+
+  const {bannerImageId, darkColor, lightColor} = currentForumEvent;
+  const pollAreaStyle: CSSProperties = {background: darkColor}
+  if (bannerImageId) {
+    const background = `top / cover no-repeat url(${makeCloudinaryImageUrl(bannerImageId, {
+      c: "fill",
+      dpr: "auto",
+      q: "auto",
+      f: "auto",
+      g: "north",
+    })}), ${darkColor}`
+    pollAreaStyle.background = background
+  }
+
+  const {ForumEventPoll} = Components;
+
+  return (
+    <AnalyticsContext pageSectionContext="forumEventPostPagePollSection">
+      <div className={classes.root} style={
+        themeOptions.name === 'dark' ? {background: darkColor, color: lightColor} : {background: lightColor, color: darkColor}
+      }>
+        <h2 className={classes.heading}>
+          {!hasVoted ? 'Have you voted yet?' : 'Did this post change your mind?'}
+        </h2>
+        <div className={classes.description}>
+          This post is part of <Link to={announcementPostUrl}>{currentForumEvent.title}</Link>.{" "}
```

Comment:
This link is broken or me as it's being treated as a relative URL instead of absolute. We probably need to use a raw `a` tag.

Reply:
Thanks for catching that! I'll fix it in my next PR